py/test_runner.py

```diff
@@ -293,8 +301,13 @@ def run_test(args):  # pylint: disable=too-many-branches,too-many-statements
       created_pods, created_services = parse_events(events)
 
       num_expected = 0
-      for replica in results.get("spec", {}).get("replicaSpecs", []):
-        num_expected += replica.get("replicas", 0)
+      if args.tfjob_version == "v1alpha1":
+        for replica in results.get("spec", {}).get("replicaSpecs", []):
+          num_expected += replica.get("replicas", 0)
+      else:
+        for replicakey in results.get("spec", {}).get("tfReplicaSpecs", {}):
+          logging.info("replicakey: %s", results.get("spec", {}).get("tfReplicaSpecs", {}).get(replicakey, {}))
+          num_expected += results.get("spec", {}).get("tfReplicaSpecs", {}).get(replicakey, {}).get("replicas", 0)
```

Comment:
Could the code work when there is a replicaspec worker while it does not specify the replicas (In this case the replicas will be set to 1)?

Reply:
Done. PTAL.